client/src/components/pages/Home/index.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import Header from './Header';
+import About from './About';
+import ReportBuilding from './ReportBuilding';
+import { Navbar } from '../../utils';
+
+class Home extends Component {
+  state = {
+    city: 'Morecambe',
+    address: '',
+    longitude: 0,
+    latitude: 0,
+  };
+
+  redirectToView = () => {
+    const { history } = this.props;
+    if (history) history.push('/view-buildings');
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+  };
+
+  handleAddressChange = (markerCoordinates, address) => {
+    if (markerCoordinates && address) {
+      const { lat: latitude, lng: longitude } = markerCoordinates;
+      this.setState({ latitude, longitude, address });
+    }
+  };
+
+  handleCityChange = ({ key }) => {
+    const city = +key === 1 ? 'Morecambe' : 'Hastings';
+    this.setState({ city });
+  };
+
+  render() {
+    const { city, latitude, longitude, address } = this.state;
+
+    return (
+      <>
+        <Navbar transparent />
+        <Header onCityChange={this.handleCityChange} />
+        <About />
+        <ReportBuilding
+          longitude={longitude}
+          latitude={latitude}
+          redirectToView={this.redirectToView}
+          city={city}
+          address={address}
+          markerCoordinates={{ lat: latitude, lng: longitude }}
+          onCityChange={this.handleCityChange}
+          handleAddressChange={this.handleAddressChange}
```

Comment:
Just a suggestion, as long as you pass all the values from the state with the exact same key names, why don't you just spread the state here...
```jsx
        <ReportBuilding
          {...this.state}
          redirectToView={this.redirectToView}
          markerCoordinates={{ lat: latitude, lng: longitude }}
          onCityChange={this.handleCityChange}
          handleAddressChange={this.handleAddressChange} 
        />
```

Reply:
i think spreading props is forbidden? https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md

Reply:
@ahmedisam99 
 when I passed it as spread it didn't work well, and it caused confusion for me !!
Did u try it in the code?

Reply:
@Amoodaa 
we're above the law :joy: